firedrake/functionspaceimpl.py

```diff
@@ -404,12 +410,14 @@ def reconstruct(self, mesh=None, name=None, **kwargs):
         if mesh is None:
             mesh = V_parent.mesh()
 
-        element = V_parent.ufl_element()
+        if element is None:
+            element = V_parent.ufl_element()
         cell = mesh.topology.ufl_cell()
         if len(kwargs) > 0 or element.cell != cell:
             element = element.reconstruct(cell=cell, **kwargs)
 
-        V = type(self).make_function_space(mesh, element, name=name)
+        V = type(self).make_function_space(mesh, element, name=name,
+                                           boundary_set=V_parent.boundary_set)
```

Comment:
@ksagiyam I want `V.reconstruct()` to preserve the `boundary_set` of the `V`. I have added the code to construct a `WithGeometry(RestrictedFunctionSpace)` above in `make_function_space`, but maybe this is not the right place.

Reply:
I think this is no longer relevant.